Automattic/jetpack
# Add full-screen dialogue component about new features in the new release

Adds the upgrade message from #7375

This PR adds a new component, `JetpackDialogue`, which acts as a wrapper for the full-screen prompt also used for the new Jumpstart.  I created it as it's own component so we can also use it for Jumpstart, but will refactor Jumpstart to use it in another PR.  

This also adds a very simple component `UpgradeNoticeContent` that will be the home of whatever description we want to show for each release.  I kept it in it's own component to make it clear where to change for each release.  

The full-screen dialogue still relies on the state() cookies, so it will still only show on the first load of the Jetpack admin after a plugin upgrade.  

Currently the full-screen dialogue is hidden behind a hard-coded check for 5.2, and will fall back to the default lame upgrade notice that is shown for any other release.  This is my attempt at a fail-safe way for us to ship this, without having to worry about it showing twice.  

**To test:** 
- manually change the `JETPACK__VERSION` constant in jetpack.php to 5.2 specifically **(NOTE: there is currently a hard-coded check to show this only for 5.2.)**
- Load the Jetapack admin UI.  You should see the full-screen dialogue about the new features. 
- Make sure you can dismiss it by clicking the `X`

Still to do: 
- [ ] Where does `Try it out!` take you? 
- [x] Add the link to the release post
- [ ] Fix remaining styles
